impl/src/main/java/org/jboss/weld/bootstrap/ExtensionBeanDeployer.java

```diff
@@ -149,8 +149,13 @@ public void addExtension(Metadata<? extends Extension> extension) {
             Set<ObserverInitializationContext<?, ?>> observerMethodInitializers, boolean isAsync) {
         ObserverMethodImpl<T, X> observer = ObserverFactory.create(method, declaringBean, beanManager, isAsync);
         ObserverInitializationContext<T, X> observerMethodInitializer = ObserverInitializationContext.of(observer, method);
-        if (isAsync && Observers.CONTAINER_LIFECYCLE_EVENT_TYPES.contains(Reflections.getRawType(observer.getObservedType()))) {
-            throw EventLogger.LOG.asyncContainerLifecycleEventObserver(observer, Formats.formatAsStackTraceElement(method.getJavaMember()));
+        if (Observers.CONTAINER_LIFECYCLE_EVENT_TYPES.contains(Reflections.getRawType(observer.getObservedType()))) {
```

Comment:
Actually, I think we should replace `CONTAINER_LIFECYCLE_EVENT_TYPES.contains()` with `Observers.isContainerLifecycleObserverMethod()`...